docs/reference/index-modules/index-sorting.asciidoc

```diff
@@ -104,10 +104,111 @@ Index sorting supports the following settings:
+[float]
+[[early-terminate]]
+=== Early termination of search request
+
+By default in elasticsearch a search request must visit every document that match a query to
+retrieve the top documents sorted by a specified sort.
+Though when the index sort and the search sort are the same it is possible to limit
+the number of documents that should be visited per segment to retrieve the N top ranked documents globally.
+For example, let's say we have an index that contains events sorted by a timestamp field:
+
+[source,js]
+--------------------------------------------------
+PUT events
+{
+    "settings" : {
+        "index" : {
+            "sort.field" : "timestamp",
+            "sort.order" : "desc" <2>
+        }
+    },
+    "mappings": {
+        "doc": {
+            "properties": {
+                "timestamp": {
+                    "type": "date"
+                }
+            }
+        }
+    }
+}
+--------------------------------------------------
+// CONSOLE
+
+<1> This index is sorted by timestamp in descending order (most recent first)
+
+You can search for the last 10 events with:
+
+[source,js]
+--------------------------------------------------
+GET /events/_search
+{
+    "size": 10,
+    "sort": [
+        { "timestamp": "desc" }
+    ]
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+Elasticsearch will detect that the top docs of each segment are already sorted in the index
+and will only compare the first N documents per segment.
+The rest of the documents matching the query are collected to count the total number of results
+and to build aggregations.
+
+If you're only looking for the last 10 events and have no interest in
+the total number of documents that match the query you can set `track_total_hits`
+to false:
+
+[source,js]
+--------------------------------------------------
+GET /events/_search
+{
+    "size": 10,
+    "sort": [
+        { "timestamp": "desc" }
+    ],
+    "track_total_hits": false
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+<1> The index sort will be used to rank the top documents and each segment will early terminate the collection after the first 10 matches.
+
+This time, Elasticsearch will not try to count the number of documents and will be able to terminate the query
+as soon as N documents have been collected per segment.
+
+[source,js]
+--------------------------------------------------
+{
+  "_shards": ...
+   "hits" : {
+      "total" : -1,     <1>
+      "max_score" : null,
+      "hits" : []
+  },
+  "took": 20,
+  "terminated_early": true,     <2>
+  "timed_out": false
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"_shards": \.\.\./"_shards": "$body._shards",/]
+// TESTRESPONSE[s/"total": \.\.\./"total": "$body.hits.total",/]
+// TESTRESPONSE[s/"took": 20,/"took": "$body.took",/]
+// TESTRESPONSE[s/"terminated_early": true,//]
```

Comment:
Not needed for `total`, thanks. 
The index is empty so `terminated_early` is false in this example.